src/provider/mod.rs

```diff
@@ -206,3 +207,89 @@ impl Provider for StackableProvider {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use indoc::indoc;
+    use rstest::rstest;
+
+    #[test]
+    fn try_to_get_package_from_complete_configuration() {
+        let pod = parse_pod_from_yaml(indoc! {"
+            apiVersion: v1
+            kind: Pod
+            metadata:
+              name: test
+            spec:
+              containers:
+              - name: kafka
+                image: kafka:2.7
+        "});
+
+        let maybe_package = StackableProvider::get_package(&pod);
+
+        if let Ok(package) = maybe_package {
+            assert_eq!("kafka", package.product);
+            assert_eq!("2.7", package.version);
+        } else {
+            panic!("Package expected but got {:?}", maybe_package);
+        }
+    }
+
+    #[rstest(pod_config, expected_err,
```

Comment:
This is fabulous!

(should you update the string on top you need to update the fixtures as well)

Reply:
Fixtures updated